docs(appmesh): note that virtual nodes allow at most one listener at this time

misterjoshua opened the pull request:
This PR adds a note that at most one listener may be added to a Virtual Node until App Mesh team adds support for multiple listeners per aws/aws-app-mesh-roadmap#120

Closes #13549

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: dff8e19978e179e6521df98f064a24ce452952b9
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f699a13f-a650-4feb-8b3c-9fb2afdbe565%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 4c13b40d4f10eabfac4935dd85a79a2db639e4da
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4c430af3-aba5-4f83-958e-5fe181a6cd92%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

skinny85 reviewed (dismissed):
Thanks for the contribution @misterjoshua!

Unfortunately, I don't think we can accept it in the current shape. The reason is that the AppMesh team is planning to lift this restriction in the future ([see here](https://github.com/aws/aws-app-mesh-roadmap/issues/120)), and thus we don't want to force AppMesh customers to update their CDK version to use the new capability. Otherwise, we could have modeled `listeners: Listener[]` as `listener: Listener`, and make it safe at compile-time!

What do you think about changing the documentation instead, to surface this limitation, and the fact that it might be lifted in the future?

Thanks,
Adam

misterjoshua commented:
> Thanks for the contribution @misterjoshua!
> 
> Unfortunately, I don't think we can accept it in the current shape. The reason is that the AppMesh team is planning to lift this restriction in the future ([see here](https://github.com/aws/aws-app-mesh-roadmap/issues/120)), and thus we don't want to force AppMesh customers to update their CDK version to use the new capability. Otherwise, we could have modeled `listeners: Listener[]` as `listener: Listener`, and make it safe at compile-time!
> 
> What do you think about changing the documentation instead, to surface this limitation, and the fact that it might be lifted in the future?
> 
> Thanks,
> Adam

@skinny85 No problem. I've reverted the behaviour changes and added a note to the `addListener()` JSDoc. Is this more acceptable?

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 404386471e040124970969c36db7cb1e3eba1100
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4686272a-66a2-4d29-b131-2649c82fbb92%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

skinny85 reviewed (approved):
Looks great @misterjoshua!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 9ad0fce5b28b38d84b1f6fb60b898830688cd0fa
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=631d2a9e-72a2-491c-98de-7bc1dc265ffc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
